### PR TITLE
[kotlinCleanUp] ModelFieldEditor.kt

### PR DESCRIPTION
**Pull request by islam-yousry**
## Purpose / Description
kotlinCleanUp of ModelFieldEditor.kt

## Fixes
Fixes part of #10489 

## Approach
Remove Hungarian notation
Use scope functions

## How Has This Been Tested?
Android Device: Samsung a03s
Run all the unit test

## Checklist
_Please, go through these checks before submitting the PR._

- [X] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [X] You have a descriptive commit message with a short title (first line, max 50 chars).
- [X] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [X] You have commented your code, particularly in hard-to-understand areas
- [X] You have performed a self-review of your own code
- [X] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [X] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/ModelFieldEditor.kt [hunk 1]
```diff
@@ -63,37 +63,35 @@ import java.lang.RuntimeException
...
-    private var mNoteFields: JSONArray? = null
-    private var mMod: Model? = null
-    private var mContextMenu: ModelEditorContextMenu? = null
-    private var mFieldNameInput: EditText? = null
-    private val mConfirmDialogCancel = Runnable { dismissContextMenu() }
+    private var currentPos = 0
+    private var fieldLabelView: ListView? = null
+    private var fieldLabels: List<String>? = null
+    private var progressDialog: MaterialDialog? = null
+    private var collectionCol: Collection? = null
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/ModelFieldEditor.kt:72]: Either `collection` or `col`, up to you!

**Review by david-allison — CHANGES_REQUESTED** [review]
Looks great, thanks!

One small change and we're good to go

**Review by david-allison — APPROVED** [review]
Looks great, thanks!!

**Review by mikehardy — APPROVED** [review]
Nice! The separate commits actually is a great demonstration of specific / useful kotlin language features

**Comment by mikehardy** [comment]
Hi there! APRIL OpenCollective Notice

Just a friendly notice that we try to process OpenCollective payments monthly - it's time for [April 2022 submissions](https://github.com/ankidroid/Anki-Android/pulls?q=is%3Apr+merged%3A2022-04-01..2022-04-30)

If you are interested in compensation for this work, the process with details is here:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process#how-to-get-paid

I only post one comment per person to avoid spamming you, regardless of the number of PRs merged, but this note applies to all PRs merged for in the month of March.

Please note that GSoC contributions are okay for this process. Our philosophy is that our users have donated to AnkiDroid for all contributions. The only PRs that will not go through the OpenCollective process are ones directly related to am accepted GSoC project from a selected participant, since those receive a stipend from GSoC itself.

Please understand that this means the monthly budget we have for payments might be stretched a bit and the cap on payments-per-person may be lower, but we try to make our process as fair and transparent as possible, we just need your understanding.

Thanks!

**Pull request merged**